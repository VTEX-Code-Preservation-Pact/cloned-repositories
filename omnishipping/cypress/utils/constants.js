export const BASE_URL = 'http://vtexgame1.vtexlocal.com.br:3000/'
export const BASE_URL_COMPLETE = 'http://vtexgame1.vtexlocal.com.br/'
export const ADD_SKUS_ENDPOINT = 'checkout/cart/add?'
export const CHECKOUT_ENDPOINT = 'checkout/'
export const PROFILE_ENDPOINT = 'api/checkout/pub/profiles/'
