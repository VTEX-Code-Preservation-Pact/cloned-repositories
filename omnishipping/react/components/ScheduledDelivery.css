.date {
  display: flex;
  align-items: center;
  position: relative;
}

.datePopper {
  z-index: 10 !important;
}

.dateLinkModify {
  border: none;
  background: transparent;
  cursor: pointer;
  color: #08c;
  font-size: 13px;
  outline: none;
  margin: 0;
  padding: 2px 0 4px;
}

.dateLink {
  border: 1px solid #357edd;
  border-radius: 4px;
  color: #357edd;
  display: block;
  font-size: 12px;
  margin: 10px 0;
  padding: 5px 8px;
  text-align: center;
  -moz-border-radius: 5px;
  -webkit-border-radius: 5px;
}

@media (max-width: 490px) {
  .dateLink {
    width: calc(100vw - 50px);
  }
}

@media (min-width: 491px) and (max-width: 766px) {
  .dateLink {
    width: calc(100vw - 100px);
  }
}

@media (min-width: 767px) {
  .dateLink {
    width: 270px;
  }
}

.dateLinkInvalid {
  border: 1px solid red;
  color: red;
}

.selectedDate {
  display: inline-block;
  margin-right: 4px;
  margin: 10px 4px 10px 0;
}

.dateLink,
.dateLink:hover,
.dateLink:active {
  background: none;
}

.scheduledDelivery {
  display: flex;
  flex-direction: column;
  padding: 10px 12px;
}

.scheduledDeliveryTitle {
  font-weight: 500;
  margin-bottom: 4px;
}

.deliveryWindowPrice {
  margin-left: 10px;
}
