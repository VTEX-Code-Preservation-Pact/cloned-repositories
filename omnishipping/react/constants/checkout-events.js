export const ENABLE = 'enable.vtex'
export const DISABLE = 'disable.vtex'
export const COMPONENT_VALIDATED = 'componentValidated.vtex'
export const COMPONENT_DONE = 'componentDone.vtex'
export const SHIPPING_INFO_UPDATED = 'shippingInformationUpdated.vtex'
export const ORDER_FORM_UPDATED = 'orderFormUpdated.vtex'
export const SHOW_MESSAGE = 'showMessage.vtex'
export const DELIVERY_SELECTED = 'deliverySelected.vtex'
export const AUTHENTICATED_USER = 'authenticatedUser.vtexid'
export const CLOSED_VTEXID = 'closed.vtexid'
