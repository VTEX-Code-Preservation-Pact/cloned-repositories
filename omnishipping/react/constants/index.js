export const PICKUP_IN_STORE = 'pickup-in-point'
export const DELIVERY = 'delivery'
export const PICKUP = 'pickup'
export const SCHEDULED_DELIVERY = 'Entrega Agendada'

export const RESIDENTIAL = 'residential'
export const COMMERCIAL = 'commercial'
export const SEARCH = 'search'
export const GIFT_REGISTRY = 'giftRegistry'

export const GEOCORDS = 'geoCoords'
export const ACCOUNT_INFO_ENDOPOINT =
  '/api/logistics/pub/shipping/configuration'
export const HIDE_MAP = 'HIDE_MAP'
export const SHOW_MAP = 'SHOW_MAP'
export const AVAILABLE = 'available'
export const SHIPPING_HASH = '#/shipping'
export const PAYMENT_HASH = '#/payment'

export const CHEAPEST = 'CHEAPEST'
export const FASTEST = 'FASTEST'
export const COMBINED = 'COMBINED'

export const DENIED = 'denied'

export const VTEXLOCAL = 'vtexlocal'
export const VTEXBETA = 'vtexcommercebeta'
export const VTEXSTABLE = 'vtexcommercestable'

export const DEFAULT_WIDTH = 50
export const DEFAULT_HEIGHT = 50

export const IMPORTANT = 'Important'
export const CRITICAL = 'Critical'
export const ERROR = 'Error'
export const INFO = 'Info'

export const OMNISHIPPING = 'omnishipping'
