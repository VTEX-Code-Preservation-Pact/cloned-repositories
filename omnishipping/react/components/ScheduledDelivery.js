import 'react-datepicker/dist/react-datepicker-cssmodules.css'

import React, { Component } from 'react'
import { injectIntl, intlShape } from 'react-intl'

import ButtonLink from './ButtonLink'
import DatePicker from 'react-datepicker'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { formatCurrency } from '../utils/Currency'
import moment from 'moment'
import { selectDeliveryWindow } from '../actions/order-form-actions'
import styles from './ScheduledDelivery.css'
import { translate } from './../utils/i18nUtils'
import uniqBy from 'lodash/uniqBy'
import get from 'lodash/get'

class ScheduledDelivery extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      selectedWindow: null,
      windowOptions: [],
      datesIncluded: [],
      outerSla: props.sla,
      locale: props.intl.locale.toLowerCase(),
    }
  }

  componentDidMount() {
    const locale = this.props.intl.locale.toLowerCase()

    import(`moment/locale/${locale}`)
      .then(() => this.initialSetup())
      .catch(error => {
        if (process.env.NODE_ENV !== 'production') {
          console.warn(error)
        }
        return import(`moment/locale/${this.getBaseLocale(locale)}`)
          .then(() => this.initialSetup())
          .catch(error => {
            if (process.env.NODE_ENV !== 'production') {
              console.warn(error)
            }
            this.initialSetup()
          })
      })
  }

  componentDidUpdate() {
    if (
      this.props.triedCompleteOmnishipping &&
      !this.state.selectedDate &&
      this.props.previousHasSelectedWindow
    ) {
      this.openCalendar()
    }
  }

  openCalendar = () => {
    const isMobile = window.innerWidth < 768

    if (isMobile) {
      document
        .querySelector(`.scheduled-delivery-choose-${this.props.sla.id}`)
        .scrollIntoView()
    }

    this._calendar.setOpen(true)
  }

  getBaseLocale(locale) {
    return locale.indexOf('-') !== -1 ? locale.split('-')[0] : locale
  }

  getSelectedDeliveryWindow = sla => {
    return sla && sla.deliveryWindow
  }

  initialSetup = () => {
    const { sla } = this.props

    this.setState({
      sla,
      datesIncluded: uniqBy(
        sla.availableDeliveryWindows.map(windowOption => ({
          date: moment.utc(windowOption.startDateUtc).format('L'),
          startDateUtc: moment
            .utc(windowOption.startDateUtc)
            .locale(this.state.locale),
        })),
        'date'
      ),
    })
    this.setDates({ outerSla: sla })
  }

  handleDatePickerChange = date => {
    this.setDates({ date, outerSla: this.props.sla })
  }

  handleChange = event => {
    const { sla } = this.state

    const selectedWindow = sla.availableDeliveryWindows
      .filter(
        windowOption =>
          windowOption.startDateUtc.toString() === event.target.value
      )
      .find(item => item)

    if (selectedWindow) {
      this.props.selectDeliveryWindow(sla && sla.id, selectedWindow)
    }

    this.setState({ selectedWindow })
  }

  setDates = ({ date, outerSla }) => {
    const { sla, selectedDate } = this.state
    const selectedDeliveryWindow = this.getSelectedDeliveryWindow(outerSla)

    const localDate =
      date || (selectedDeliveryWindow && selectedDeliveryWindow.startDateUtc)
    const availableWindowsDay =
      sla &&
      sla.availableDeliveryWindows &&
      sla.availableDeliveryWindows.filter(
        windowOption =>
          moment(windowOption.startDateUtc).format('YYYY MM DD') ===
          moment(localDate).format('YYYY MM DD')
      )

    if (!selectedDeliveryWindow || localDate !== selectedDate) {
      const deliveryWindow =
        (availableWindowsDay && availableWindowsDay[0]) === undefined
          ? null
          : availableWindowsDay[0]
      deliveryWindow &&
        this.props.selectDeliveryWindow(sla && get(sla, 'id'), deliveryWindow)
    }

    this.setState({
      selectedDate: localDate,
      windowOptions: availableWindowsDay,
      selectedWindow: selectedDeliveryWindow || availableWindowsDay[0],
    })
  }

  render() {
    const {
      datesIncluded,
      locale,
      selectedWindow,
      selectedDate,
      windowOptions,
      sla,
    } = this.state

    const { intl, storePreferencesData } = this.props

    const hasSelectedDateAndWindow =
      !!selectedDate && !!selectedWindow && windowOptions.length > 0

    const hasSelectedDateWithoutWindows =
      !!selectedDate && windowOptions.length === 0

    return (
      <div className={styles.scheduledDelivery}>
        <span className={styles.date}>
          <DatePicker
            ref={c => (this._calendar = c)}
            className={styles.datepicker}
            customInput={
              <div>
                <span
                  className={`${
                    selectedDate ? styles.selectedDate : ''
                  } shp-selected-date`}>
                  {selectedDate &&
                    `${moment(selectedDate)
                      .locale(locale)
                      .format(translate(intl, 'datepickerDateFormat'))}`}
                </span>
                {selectedDate ? (
                  <ButtonLink
                    style={`${
                      styles.dateLinkModify
                    } shp-datepicker-modify-button`}>
                    {translate(intl, 'modify')}
                  </ButtonLink>
                ) : (
                  <ButtonLink
                    slaId={sla && sla.id}
                    id="scheduled-delivery-choose"
                    className="scheduled-delivery-choose"
                    style={`${styles.dateLink} shp-datepicker-button`}>
                    {translate(intl, 'scheduledDeliveryChoice')}
                  </ButtonLink>
                )}
              </div>
            }
            dateFormat={translate(intl, 'datepickerDateFormat')}
            includeDates={datesIncluded.map(date => date.startDateUtc)}
            locale={locale}
            onChange={this.handleDatePickerChange}
            popperClassName={styles.datePopper}
            popperPlacement={'top'}
            selected={selectedDate && moment(selectedDate).locale(locale)}
          />
        </span>
        {sla &&
          hasSelectedDateAndWindow && (
          <select
            className="span12"
            id={`scheduled-delivery-${sla.id}`}
            onChange={this.handleChange}
            value={selectedWindow.startDateUtc}>
            {windowOptions.map(windowOption => {
              const startDate = moment
                .utc(windowOption.startDateUtc)
                .format('LT')

              const endDate = moment.utc(windowOption.endDateUtc).format('LT')

              const formattedPrice = formatCurrency({
                value: windowOption.price + sla.price,
                storePreferencesData: storePreferencesData,
              })
              return (
                <option
                  id={windowOption.startDateUtc}
                  key={windowOption.startDateUtc + windowOption.endDateUtc}
                  value={windowOption.startDateUtc}>
                  {translate(intl, 'deliveryWindows', {
                    startDate,
                    endDate,
                  })}
                    &nbsp;&nbsp;&nbsp;&nbsp;
                  {windowOption.price + sla.price === 0
                    ? translate(intl, 'free')
                    : formattedPrice}
                </option>
              )
            })}
          </select>
        )}
        {sla &&
          hasSelectedDateWithoutWindows && (
          <p>{translate(intl, 'noTimesAvailable')}</p>
        )}
      </div>
    )
  }
}

ScheduledDelivery.propTypes = {
  intl: intlShape,
  previousHasSelectedWindow: PropTypes.bool,
  selectDeliveryWindow: PropTypes.func.isRequired,
  sla: PropTypes.object,
  storePreferencesData: PropTypes.object,
  triedCompleteOmnishipping: PropTypes.bool,
}

const mapStateToProps = state => ({
  storePreferencesData: state.orderForm.storePreferencesData,
  triedCompleteOmnishipping: state.componentActivity.triedCompleteOmnishipping,
})

export default connect(
  mapStateToProps,
  {
    selectDeliveryWindow,
  }
)(injectIntl(ScheduledDelivery))
