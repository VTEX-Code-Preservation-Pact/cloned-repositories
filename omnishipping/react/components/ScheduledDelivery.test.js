import ScheduledDelivery from './ScheduledDelivery'
import { DELIVERY, SCHEDULED_DELIVERY, CHEAPEST } from '../constants'
import { Provider } from 'react-redux'

import { IntlProvider } from 'react-intl'
import { mount, shallow } from 'enzyme'

import messages from '../../messages/pt.json'

import MockDate from 'mockdate'
import React from 'react'

import 'moment-timezone'
import moment from 'moment'

moment.tz.setDefault('America/Los_Angeles')

describe('ScheduledDelivery', () => {
  let state, store, props

  beforeEach(() => {
    state = {
      componentActivity: {
        triedCompleteOmnishipping: false,
      },
      delivery: {
        activeDeliveryOption: CHEAPEST,
        CHEAPEST: [
          {
            slas: [
              {
                deliveryChannel: DELIVERY,
                name: SCHEDULED_DELIVERY,
                price: 1,
                shippingEstimate: '0bd',
                id: SCHEDULED_DELIVERY,
                availableDeliveryWindows: [
                  {
                    endDateUtc: '2017-07-14T00:00:00+00:00',
                    lisPrice: 200,
                    price: 200,
                    startDateUtc: '2017-07-14T18:00:00+00:00',
                    tax: 20,
                  },
                  {
                    endDateUtc: '2017-07-13T00:00:00+00:00',
                    lisPrice: 200,
                    price: 200,
                    startDateUtc: '2017-07-13T18:00:00+00:00',
                    tax: 20,
                  },
                  {
                    endDateUtc: '2017-07-15T00:00:00+00:00',
                    lisPrice: 200,
                    price: 200,
                    startDateUtc: '2017-07-15T18:00:00+00:00',
                    tax: 20,
                  },
                  {
                    endDateUtc: '2017-07-16T00:00:00+00:00',
                    lisPrice: 200,
                    price: 200,
                    startDateUtc: '2017-07-16T18:00:00+00:00',
                    tax: 20,
                  },
                ],
                deliveryWindow: {
                  endDateUtc: '2017-07-14T18:00:00+00:00',
                  lisPrice: 200,
                  price: 200,
                  startDateUtc: '2017-07-14T15:00:00+00:00',
                  tax: 20,
                },
              },
            ],
          },
        ],
      },
      orderForm: {
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
        shippingData: {
          logisticsInfo: [
            {
              slas: [
                {
                  deliveryChannel: DELIVERY,
                  name: SCHEDULED_DELIVERY,
                  price: 1,
                  shippingEstimate: '0bd',
                  id: SCHEDULED_DELIVERY,
                  availableDeliveryWindows: [
                    {
                      endDateUtc: '2017-07-14T00:00:00+00:00',
                      lisPrice: 200,
                      price: 200,
                      startDateUtc: '2017-07-14T18:00:00+00:00',
                      tax: 20,
                    },
                    {
                      endDateUtc: '2017-07-13T00:00:00+00:00',
                      lisPrice: 200,
                      price: 200,
                      startDateUtc: '2017-07-13T18:00:00+00:00',
                      tax: 20,
                    },
                    {
                      endDateUtc: '2017-07-15T00:00:00+00:00',
                      lisPrice: 200,
                      price: 200,
                      startDateUtc: '2017-07-15T18:00:00+00:00',
                      tax: 20,
                    },
                    {
                      endDateUtc: '2017-07-16T00:00:00+00:00',
                      lisPrice: 200,
                      price: 200,
                      startDateUtc: '2017-07-16T18:00:00+00:00',
                      tax: 20,
                    },
                  ],
                  deliveryWindow: {
                    endDateUtc: '2017-07-14T18:00:00+00:00',
                    lisPrice: 200,
                    price: 200,
                    startDateUtc: '2017-07-14T15:00:00+00:00',
                    tax: 20,
                  },
                },
              ],
            },
          ],
        },
      },
      windowOptions: [
        {
          startDateUtc: '2017-07-14T18:00:00+00:00',
          endDateUtc: '2017-07-14T00:00:00+00:00',
          lisPrice: 100,
        },
      ],
      selectedWindow: {
        endDateUtc: '2017-07-14T12:00:00+00:00',
        lisPrice: 200,
        price: 200,
        startDateUtc: '2017-07-14T06:00:00+00:00',
        tax: 20,
      },
      selectedDate: '2017-07-14T00:00:00+00:00',
      minDate: {
        endDateUtc: '2017-07-14T00:00:00+00:00',
        lisPrice: 200,
        price: 200,
        startDateUtc: '2017-07-14T18:00:00+00:00',
        tax: 20,
      },
      maxDate: {
        endDateUtc: '2017-07-14T00:00:00+00:00',
        lisPrice: 200,
        price: 200,
        startDateUtc: '2017-07-14T18:00:00+00:00',
        tax: 20,
      },
    }
    store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn(() => state),
    }
    props = {
      selectDeliveryWindow: jest.fn(),
      storePreferencesData: {
        countryCode: 'BRA',
        currencyCode: 'BRL',
        currencySymbol: 'R$',
        currencyFormatInfo: {
          currencyDecimalDigits: 2,
          currencyDecimalSeparator: ',',
          currencyGroupSeparator: '.',
          currencyGroupSize: 3,
          startsWithCurrencySymbol: true,
        },
      },
      index: 0,
      sla: {
        deliveryChannel: DELIVERY,
        name: SCHEDULED_DELIVERY,
        price: 1,
        shippingEstimate: '0bd',
        id: SCHEDULED_DELIVERY,
        availableDeliveryWindows: [
          {
            endDateUtc: '2017-07-14T00:00:00+00:00',
            lisPrice: 200,
            price: 200,
            startDateUtc: '2017-07-14T18:00:00+00:00',
            tax: 20,
          },
          {
            endDateUtc: '2017-07-13T00:00:00+00:00',
            lisPrice: 200,
            price: 200,
            startDateUtc: '2017-07-13T18:00:00+00:00',
            tax: 20,
          },
          {
            endDateUtc: '2017-07-15T00:00:00+00:00',
            lisPrice: 200,
            price: 200,
            startDateUtc: '2017-07-15T18:00:00+00:00',
            tax: 20,
          },
          {
            endDateUtc: '2017-07-16T00:00:00+00:00',
            lisPrice: 200,
            price: 200,
            startDateUtc: '2017-07-16T18:00:00+00:00',
            tax: 20,
          },
        ],
        deliveryWindow: {
          endDateUtc: '2017-07-14T18:00:00+00:00',
          lisPrice: 200,
          price: 200,
          startDateUtc: '2017-07-14T15:00:00+00:00',
          tax: 20,
        },
      },
    }
  })

  it('should render self and subcomponents', done => {
    MockDate.set(Date.UTC(2017, 6, 14, 18, 0, 0, 0), 180)
    const wrapper = shallow(
      <Provider store={store}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <ScheduledDelivery {...props} />
        </IntlProvider>
      </Provider>
    )

    process.nextTick(() => {
      try {
        wrapper.update()
        expect(wrapper.getElement()).toMatchSnapshot()
      } catch (e) {
        return done(e)
      }
      done()
    })
  })

  it('should trigger handleRadioChange clicking a time', done => {
    MockDate.set(Date.UTC(2017, 6, 14, 18, 0, 0, 0), 180)

    const wrapper = mount(
      <Provider store={store}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <ScheduledDelivery {...props} store={store} />
        </IntlProvider>
      </Provider>
    )

    process.nextTick(() => {
      try {
        wrapper.update()
        const event = { target: { value: '2017-07-14T18:00:00+00:00' } }

        wrapper.find('input[type="radio"]').simulate('change', event)

        expect(wrapper.state('selectedWindow')).toMatchSnapshot()
      } catch (e) {
        return done(e)
      }
      done()
    })
  })
})
