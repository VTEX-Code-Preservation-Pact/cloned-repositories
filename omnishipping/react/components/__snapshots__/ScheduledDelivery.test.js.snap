// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduledDelivery should render self and subcomponents 1`] = `
<IntlProvider
  locale="pt"
  messages={
    Object {
      "country.BRA": "BRA",
      "omnishipping.addDeliveryPlaceButton": "Entregar em outro endereço",
      "omnishipping.aditionalInfo": "Outras informações",
      "omnishipping.allItemsPickup": "Apenas a opção de Retirar em um Ponto está disponível para seus itens.",
      "omnishipping.askGeolocation": "Pesquise um endereço com sua localização",
      "omnishipping.askGeolocationAccept": "Usar minha localização atual",
      "omnishipping.askGeolocationDenied": "Buscar por um endereço",
      "omnishipping.askGeolocationDeniedSubtitle": "Procure por endereços que você frequenta e vamos localizar os pontos próximos",
      "omnishipping.askGeolocationDeniedTitle": "Encontre pontos de retirada próximos",
      "omnishipping.askGeolocationInstructions": "Selecione \\"Permitir\\" no alerta exibido no seu navegador",
      "omnishipping.askGeolocationManual": "Procurar endereço manualmente",
      "omnishipping.askGeolocationSearching": "Buscando sua localização",
      "omnishipping.askGeolocationSubtitle": "Para isso, precisamos saber sua localização",
      "omnishipping.askGeolocationTitle": "Vamos sugerir o ponto de retirada mais perto de você",
      "omnishipping.askGeolocationWaiting": "Aguardando sua autorização",
      "omnishipping.awatingData": "Aguardando o preenchimento dos dados",
      "omnishipping.backToAddressList": "Voltar para lista de endereços",
      "omnishipping.cancelBackList": "Consultar outros pontos",
      "omnishipping.cancelPickup": "Cancelar retirada",
      "omnishipping.change": "Ver todos os pontos de retirada",
      "omnishipping.changePostalCode": "Altere seu CEP",
      "omnishipping.cheapestOption": "Mais econômica",
      "omnishipping.chooseDelivery": "Receber em",
      "omnishipping.choosePickup": "Pacote e prazo",
      "omnishipping.closeButton": "fechar",
      "omnishipping.combinedOption": "Combinada",
      "omnishipping.confirmPoint": "Retirar neste ponto",
      "omnishipping.datepickerDateFormat": "dddd, LL",
      "omnishipping.deliver": "Receber",
      "omnishipping.deliverComplement": " ",
      "omnishipping.deliveryAddressTitle": "Endereço de entrega",
      "omnishipping.deliveryDate": "Agendamento",
      "omnishipping.deliveryOption": "Forma de entrega",
      "omnishipping.deliveryOptionsLeft": "Forma de entrega para os itens restantes",
      "omnishipping.deliverySplitDisclaimer": "Alguns itens estão indisponíveis para o seu endereço. Retire-os em um ponto.",
      "omnishipping.deliverySplitDisclaimerLogged": "Alguns itens estão indisponíveis para o seu endereço. Retire-os em um ponto.",
      "omnishipping.deliverySummary": "Receber",
      "omnishipping.deliveryTime": "Hora da entrega",
      "omnishipping.deliveryTitle": "Entrega",
      "omnishipping.deliveryWindows": "Das {startDate} às {endDate}",
      "omnishipping.deliveryWindowsSummary": "{date}, entre {startDate} e {endDate}",
      "omnishipping.deliveryWindowsSummarytime": "entre {startDate} e {endDate}",
      "omnishipping.details": "Ver mais detalhes",
      "omnishipping.editSelectedAddressButton": "Modificar endereço selecionado",
      "omnishipping.errorId": "ErrorId: {errorId}",
      "omnishipping.fastestOption": "Mais rápida",
      "omnishipping.fillDeliveryData": "Preencher Entrega",
      "omnishipping.findPickup": "Encontre um ponto de retirada perto de você.",
      "omnishipping.free": "Grátis",
      "omnishipping.items": "{itemsAmount, plural, =0 {} one {# item} other {# itens}}",
      "omnishipping.jsError": "Algo inesperado aconteceu",
      "omnishipping.list": "Lista",
      "omnishipping.loading": "Carregando…",
      "omnishipping.locatePickupPoint": "Localizar um ponto",
      "omnishipping.map": "Mapa",
      "omnishipping.modify": "alterar",
      "omnishipping.modifyDeliveryOptions": "Alterar opções de entrega",
      "omnishipping.modifyPostalCode": "alterar",
      "omnishipping.noAddressAction": "confirme sua identidade",
      "omnishipping.noAddressCTA": "Para adicionar um endereço, {action}.",
      "omnishipping.noAddressTitle": "Dados de endereço não preenchidos.",
      "omnishipping.noPickupSelected": "Nenhuma loja selecionada",
      "omnishipping.noTimesAvailable": "Nenhum horário disponível para esta data.",
      "omnishipping.of": "de",
      "omnishipping.onePackageIn": "Um pacote em",
      "omnishipping.optionPackagesDetails": "{items, plural, =0 {...} one {{shippingEstimate}} other {{items} pacotes em prazos variados}}",
      "omnishipping.package": "Pacote",
      "omnishipping.paymentCTA": "Ir para o pagamento",
      "omnishipping.pickUpItems": "{items, plural, =0 {} one {Retirar item em um ponto} other {Retirar itens em um ponto}}",
      "omnishipping.pickup": "Retirar",
      "omnishipping.pickupComplement": " ",
      "omnishipping.pickupOption": "Forma de retirada",
      "omnishipping.pickupPoint.modify": "Alterar",
      "omnishipping.pickupReceiver": "Responsável pela retirada",
      "omnishipping.pickupSomeItems": "1. Escolher opções de Retirada",
      "omnishipping.pickupSplitDisclaimer": "Alguns itens estão indisponíveis no ponto selecionado. Preencha seus dados de entrega para recebê-los.",
      "omnishipping.pickupSplitDisclaimerLogged": "Alguns itens estão indisponíveis neste ponto. Preencha seus dados de entrega para recebê-los.",
      "omnishipping.pickupSummary": "Retirar",
      "omnishipping.pointDetails": "Detalhes do ponto",
      "omnishipping.price": "{value, plural, =0 {Grátis} other {{formattedPrice}}}",
      "omnishipping.productsFromSeller": "Produtos de {seller}",
      "omnishipping.productsInPoint": "Itens disponíveis neste ponto",
      "omnishipping.receiveRemainingPackages": "2. Receber produtos restantes",
      "omnishipping.reloadThePage": "Recarregue a página",
      "omnishipping.removeFromCart": "{items, plural, =0 {} one {remova o item do carrinho.} other {remova os items do carrinho.}}",
      "omnishipping.scheduled": "Agendada",
      "omnishipping.scheduledDeliveryAll": "Agendar todos",
      "omnishipping.scheduledDeliveryChoice": "Escolher data e hora de entrega",
      "omnishipping.scheduledDeliveryPartial": "{amount, plural, =0 {} one {Agendar # item} other {Agendar # itens}}",
      "omnishipping.scheduledDeliveryPrice": "A partir de {price}",
      "omnishipping.scheduledDeliverySplitDisclaimer": "Alguns itens não podem ser agendados. Selecione uma forma de entrega para recebê-los.",
      "omnishipping.searchLocationMap": "Buscar por Lugar, CEP, Bairro ou Cidade",
      "omnishipping.selectDeliveryOptions": "Selecione as formas de entrega",
      "omnishipping.selectPickupPoint": "Selecione um ponto de retirada",
      "omnishipping.shippingAddress": "Endereço de entrega",
      "omnishipping.shippingEstimate-bd": "{timeAmount, plural, =0 {Mesmo dia} one {Em até # dia útil} other {Em até # dias úteis}}",
      "omnishipping.shippingEstimate-d": "{timeAmount, plural, =0 {Mesmo dia} one {Em até # dia} other {Em até # dias}}",
      "omnishipping.shippingEstimate-db": "Dias úteis",
      "omnishipping.shippingEstimate-h": "{timeAmount, plural, =0 {Em alguns instantes} one {Em até # hora} other {Em até # horas}}",
      "omnishipping.shippingEstimate-m": "{timeAmount, plural, =0 {Em alguns instantes} one {Em até # minuto} other {Em até # minutos}}",
      "omnishipping.shippingEstimatePickup-bd": "{timeAmount, plural, =0 {Mesmo dia} one {A partir de # dia útil} other {A partir de # dias úteis}}",
      "omnishipping.shippingEstimatePickup-d": "{timeAmount, plural, =0 {Mesmo dia} one {A partir de # dia} other {A partir de # dias}}",
      "omnishipping.shippingEstimatePickup-h": "{timeAmount, plural, =0 {Em alguns instantes} one {A partir de # hora} other {A partir de # horas}}",
      "omnishipping.shippingEstimatePickup-m": "{timeAmount, plural, =0 {Em alguns instantes} one {A partir de # minuto} other {A partir de # minutos}}",
      "omnishipping.shippingEstimateScheduled": "Entrega agendada",
      "omnishipping.slaOption": "{shippingEstimate} - {price, plural, =0 {Grátis} other {{price}}} ({name})",
      "omnishipping.slaOptionSimple": "{shippingEstimate}{scheduled}",
      "omnishipping.unavailable": "indisponível",
      "omnishipping.unavailableDelivery": "{items, plural, =0 {} one {O item a seguir está com entrega indisponível para o CEP escolhido.} other {Os itens a seguir estão com entrega indisponível para o CEP escolhido.}}",
      "omnishipping.unavailableDeliveryAvailablePickup": "{items, plural, =0 {} one {O item a seguir está indisponível para entrega. Você pode retirá-lo em um ponto.} other {Os itens a seguir estão indisponíveis para entrega. Você pode retirá-los em um ponto.}}",
      "omnishipping.unavailableDeliveryOptions": "Escolha outro ponto de retirada ou {removeFromCart}",
      "omnishipping.unavailableDeliveryOptionsPostalCode": "Altere seu CEP ou {removeFromCart}",
      "omnishipping.unavailableItemsAmount": "{itemsAmount, plural, =0 {} one {# indisponível} other {# indisponíveis}}",
      "omnishipping.yourPackages": "Pacotes e prazos",
    }
  }
>
  <Connect(InjectIntl(ScheduledDelivery))
    index={0}
    selectDeliveryWindow={[Function]}
    sla={
      Object {
        "availableDeliveryWindows": Array [
          Object {
            "endDateUtc": "2017-07-14T00:00:00+00:00",
            "lisPrice": 200,
            "price": 200,
            "startDateUtc": "2017-07-14T18:00:00+00:00",
            "tax": 20,
          },
          Object {
            "endDateUtc": "2017-07-13T00:00:00+00:00",
            "lisPrice": 200,
            "price": 200,
            "startDateUtc": "2017-07-13T18:00:00+00:00",
            "tax": 20,
          },
          Object {
            "endDateUtc": "2017-07-15T00:00:00+00:00",
            "lisPrice": 200,
            "price": 200,
            "startDateUtc": "2017-07-15T18:00:00+00:00",
            "tax": 20,
          },
          Object {
            "endDateUtc": "2017-07-16T00:00:00+00:00",
            "lisPrice": 200,
            "price": 200,
            "startDateUtc": "2017-07-16T18:00:00+00:00",
            "tax": 20,
          },
        ],
        "deliveryChannel": "delivery",
        "deliveryWindow": Object {
          "endDateUtc": "2017-07-14T18:00:00+00:00",
          "lisPrice": 200,
          "price": 200,
          "startDateUtc": "2017-07-14T15:00:00+00:00",
          "tax": 20,
        },
        "id": "Entrega Agendada",
        "name": "Entrega Agendada",
        "price": 1,
        "shippingEstimate": "0bd",
      }
    }
    storePreferencesData={
      Object {
        "countryCode": "BRA",
        "currencyCode": "BRL",
        "currencyFormatInfo": Object {
          "currencyDecimalDigits": 2,
          "currencyDecimalSeparator": ",",
          "currencyGroupSeparator": ".",
          "currencyGroupSize": 3,
          "startsWithCurrencySymbol": true,
        },
        "currencySymbol": "R$",
      }
    }
  />
</IntlProvider>
`;
